ams/server/src/main/java/com/netease/arctic/server/table/TableMetrics.java

```diff
@@ -0,0 +1,93 @@
+package com.netease.arctic.server.table;
+
+import com.netease.arctic.ams.api.metrics.*;
+import com.netease.arctic.server.metrics.MetricRegistry;
+import com.netease.arctic.server.optimizing.OptimizingStatus;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+
+public class TableMetrics {
+  /** Table is no need optimizing. */
+  public static final String STATE_IDLE = "idle";
+
+  /** Table is need optimizing, but waiting for resource */
+  public static final String STATE_PENDING = "pending";
+
+  /** Table is doing optimizing process planing. */
+  public static final String STATE_PLANING = "planing";
+
+  /** Table is executing optimizing process */
+  public static final String STATE_EXECUTING = "executing";
+
+  /** All optimizing process task is done, and process is committing. */
+  public static final String STATE_COMMITTING = "committing";
+
+  private final ServerTableIdentifier identifier;
+
+  private String state = STATE_IDLE;
+  private long stateSetTimestamp = System.currentTimeMillis();
+
+  private boolean register = false;
+  private final MetricRegistry tableMetricRegistry = new MetricRegistry();
+
+  public TableMetrics(ServerTableIdentifier identifier) {
+    this.identifier = identifier;
+    createOptimizingMetric(
+        MetricDefines.TABLE_OPTIMIZING_STATE_IDLE_DURATION, new StateDurationGauge(STATE_IDLE));
+    createOptimizingMetric(
+        MetricDefines.TABLE_OPTIMIZING_STATE_PENDING_DURATION,
+        new StateDurationGauge(STATE_PENDING));
+    createOptimizingMetric(
+        MetricDefines.TABLE_OPTIMIZING_STATE_PLANNING_DURATION,
+        new StateDurationGauge(STATE_PLANING));
+    createOptimizingMetric(
+        MetricDefines.TABLE_OPTIMIZING_STATE_EXECUTING_DURATION,
+        new StateDurationGauge(STATE_EXECUTING));
```

Comment:
The `OptimizingStatus` of the amoro table has not defined the EXECUTING status.    


Is it necessary to have conversion   in StateDurationGauge that mapping OPTIMIZING_STATUS (= full/major/minor)  into  EXECUTING.

Reply:
fixed.